pyccel/codegen/printing/cwrappercode.py

```diff
@@ -57,49 +60,51 @@ def _print_FunctionDef(self, expr):
             if cast_func is not None:
                 # TODO: Add other properties
                 collect_var = Variable(dtype=collect_type,
-                        name=get_new_name(used_names, a.name+"_tmp"))
+                        name=self.et_new_name(used_names, a.name+"_tmp"))
                 wrapper_vars.append(collect_var)
                 parse_args.append(collect_var)
                 wrapper_body.append(cast_func(a, collect_var))
             else:
                 parse_args.append(a)
-            type_keys.append(pytype_registry[str_dtype(arg.dtype)])
+            type_keys += pytype_registry[a.dtype]
 
         # TODO: Create PyArg_ParseTupleNode
-        wrapper_body.insert(0,If((Not(PyArg_ParseTupleNode(python_func_args, type_keys, parse_args)), Return(Nil))))
+        wrapper_body.insert(0,If((PyccelNot(PyArg_ParseTupleNode(python_func_args, type_keys, parse_args)), [Return([Nil])])))
 
         if len(expr.results)==0:
             func_call = FunctionCall(expr, expr.arguments)
         else:
             results = expr.results if len(expr.results)>1 else expr.results[0]
             func_call = Assign(results,FunctionCall(expr, expr.arguments))
         wrapper_body.append(func_call)
-
         #TODO: Loop over results to carry out necessary casts and collect Py_BuildValue type string
-
+        name = expr.name.name+"_wrapper"
         #TODO: Create node and add args
-        wrapper_body.append(Return(PyBuildValueNode()))
-
-        wrapper_func = FunctionDef(name = self.parser.get_new_name(expr.name+"_wrapper"),
+        wrapper_func = FunctionDef(name = expr.name.name+"_wrapper" ,
             arguments = wrapper_args,
             results = wrapper_results,
             body = wrapper_body,
             local_vars = wrapper_vars)
-
-        return CWrapper._print_FunctionDef(wrapper_func)
+        return CCodePrinter._print_FunctionDef(self, wrapper_func)
 
     def _print_Module(self, expr):
         function_signatures = '\n'.join('{};'.format(self.function_signature(f)) for f in expr.funcs)
 
         function_defs = '\n'.join(self._print(f) for f in expr.funcs)
 
-        #TODO: Print ModuleDef (see cwrapper.py L69)
+        module_def = ('static struct PyModuleDef {mod_name}_module = {{\n'
+                '   PyModuleDef_HEAD_INIT,\n'
+                '   \"{mod_name}\",   /* name of module */\n'
+                '   NULL, /* module documentation, may be NULL */\n'
+                '   -1,       /* size of per-interpreter state of the module,*/\n'
+                '   {mod_name}_methods\n'
+                '}};\n\n'.format(mod_name = expr.name))
```

Comment:
The indentation should be added automatically. I think it's in the doprint function.
The comments can probably be turned into python comments.
I can't think of a reason to prefer c comments in the wrapper. Can you @MenoIy ?